rgw/v2/tests/s3_swift/test_bucket_listing.py

```diff
@@ -197,88 +204,106 @@ def test_exec(config, ssh_con):
                             utils.create_psuedo_dir(s3_pseudo_dir_name, bucket)
 
                 # radoslist listing of the bucket
-                if config.test_ops["radoslist"] is True:
-                    log.info("executing the command radosgw-admin bucket radoslist ")
-                    radoslist = utils.exec_shell_cmd(
-                        "radosgw-admin bucket radoslist --bucket %s"
-                        % bucket_name_to_create
-                    )
-                    if radoslist is False:
-                        raise TestExecError("Radoslist command execution failed")
+                if config.test_ops.get("create_object", None) is not None:
+                    if config.test_ops["radoslist"] is True:
+                        log.info("executing the command radosgw-admin bucket radoslist")
+                        radoslist = utils.exec_shell_cmd(
+                            "radosgw-admin bucket radoslist --bucket %s"
+                            % bucket_name_to_create
+                        )
+                        if radoslist is False:
+                            raise TestExecError("Radoslist command execution failed")
 
-                # get the configuration parameter - rgw_bucket_index_max_aio
-                ceph_version_id, ceph_version_name = utils.get_ceph_version()
-                if ceph_version_name in ["luminous", "nautilus"]:
-                    if ssh_con:
-                        cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
-                        _, stdout, _ = ssh_con.exec_command(cmd)
-                        max_aio_output = stdout.readline()
+                    # get the configuration parameter - rgw_bucket_index_max_aio
+                    ceph_version_id, ceph_version_name = utils.get_ceph_version()
+                    if ceph_version_name in ["luminous", "nautilus"]:
+                        if ssh_con:
+                            cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                            _, stdout, _ = ssh_con.exec_command(cmd)
+                            max_aio_output = stdout.readline()
+                        else:
+                            cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                            max_aio_output = utils.exec_shell_cmd(cmd)
+                        max_aio = max_aio_output.split()[1]
                     else:
-                        cmd = "ceph daemon `ls -t /var/run/ceph/ceph-client.rgw.*.asok|head -1` config show |grep  rgw_bucket_index_max_aio"
+                        cmd = "ceph config get mon rgw_bucket_index_max_aio"
```

Comment:
@ckulal , grep of rgw_bucket_index_max_aio from asok file was added for backward compatibility(nautilus), please retain the changes.

Reply:
This checks with asok file is not removed, it is still available in line no: 225